editor/editor_file_system.cpp

```diff
@@ -369,6 +384,11 @@ void EditorFileSystem::_scan_filesystem() {
 					fc.script_class_name = split[7].get_slice("<>", 0);
 					fc.script_class_extends = split[7].get_slice("<>", 1);
 					fc.script_class_icon_path = split[7].get_slice("<>", 2);
+					fc.import_md5 = split[7].get_slice("<>", 3);
+					String dest_paths = split[7].get_slice("<>", 4);
```

Comment:
With so many slices for the same separator, I wonder if it wouldn't be faster to actually use `split()` and then index the list.

Reply:
After some testing on a project with more then 60000 files, it's virtually the same performance. It took around 900ms to load the file with the get_slice and the split.
And the code is easier to read with the get_slice:
```cpp
Vector<String> other_infos = split[7].split("<>");
int size_other_infos = other_infos.size();
if (size_other_infos > 0) {
  fc.script_class_name = other_infos[0];
}
if (size_other_infos > 1) {
   fc.script_class_extends = other_infos[1];
}
if (size_other_infos > 2) {
  fc.script_class_icon_path = other_infos[2];
}
if (size_other_infos > 3) {
  fc.import_md5 = other_infos[3];
}
if (size_other_infos > 4) {
  String dest_paths = other_infos[4];
  if (!dest_paths.is_empty()) {
    fc.import_dest_paths = dest_paths.split("<*>");
  }
}
```
So, if it's ok with, you, I suggest to keep the get_slice?

Reply:
Well I would have done:
```cpp
Vector<String> other_infos = split[7].split("<>");
ERR_FAIL_COND(other_infos.size() < 5);
fc.script_class_name = other_infos[0];
fc.script_class_extends = other_infos[1];
fc.script_class_icon_path = other_infos[2];
fc.import_md5 = other_infos[3];
String dest_paths = other_infos[4];
if (!dest_paths.is_empty()) {
    fc.import_dest_paths = dest_paths.split("<*>");
}
```
which seems pretty readable to me.

But if there's no performance difference, either way is fine.